### ISSUE TITLE
ROCm 3.0: Stack smashing detected

### ISSUE THREAD

**Issue by sebpuetz** [ROCm/pytorch#559]
## 🐛 Bug

Simple operations lead to stack smashing after upgrade to ROCm 3.0 and recompilation

## To Reproduce

```
import torch
torch.zeros(100).cuda()
*** stack smashing detected ***: <unknown> terminated
[1]    10291 abort (core dumped)  ipython
```

## Expected behavior

No stack smashing

## Environment

```
Collecting environment information...
PyTorch version: 1.4.0a0+7f1a03b
Is debug build: No
CUDA used to build PyTorch: Could not collect

OS: Ubuntu 18.04.3 LTS
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
CMake version: version 3.10.2

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.17.4
[pip3] numpydoc==0.9.1
[pip3] pytorch-pretrained-bert==0.6.2
[pip3] pytorch-transformers==1.1.0
[pip3] torch==1.4.0a0+7f1a03b
[pip3] torchvision==0.5.0a0+d2c763e
[conda] Could not collect
```

 - PyTorch Version (e.g., 1.0): master branch from https://github.com/ROCmSoftwarePlatform/pytorch/ 
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source):
```
sudo apt install rock-dkms rocm-dev rocm-libs miopen-hip hipsparse rocthrust hipcub rccl roctracer-dev
git clone git@github.com:ROCmSoftwarePlatform/pytorch
cd pytorch
git submodule update --init --recursive
cd /data/pytorch/
python tools/amd_build/build_amd.py
sudo sed -i 's/find_dependency(hip)/find_dependency(HIP)/g' /opt/rocm/rocsparse/lib/cmake/rocsparse/rocsparse-config.cmake
sudo sed -i 's/find_dependency(hip)/find_dependency(HIP)/g' /opt/rocm/rocfft/lib/cmake/rocfft/rocfft-config.cmake
sudo sed -i 's/find_dependency(hip)/find_dependency(HIP)/g' /opt/rocm/miopen/lib/cmake/miopen/miopen-config.cmake
sudo sed -i 's/find_dependency(hip)/find_dependency(HIP)/g' /opt/rocm/rocblas/lib/cmake/rocblas/rocblas-config.cmake
pip install enum34 numpy pyyaml setuptools typing cffi future hypothesis
export PYTORCH_ROCM_ARCH=gfx906
USE_ROCM=1 USE_LMDB=1 USE_OPENCV=1 MAX_JOBS=14 python setup.py bdist_wheel
pip install dist/torch-1.4.0a0+97e1b3c-cp37-cp37m-linux_x86_64.whl
python -c "import torch"
```
 - Python version: 3.7
 - ROCm version: 3.0
 - GPU models and configuration: Radeon VII

**Comment by iotamudelta** [comment]
This is a bug in our stack which should be resolved in an upcoming release. Sorry!

**Comment by dodatko** [comment]
@iotamudelta Is it about current master or whole ROCm3.0 stack?

**Comment by sebpuetz** [comment]
The docker image works fine tho:

```Bash
alias drun='docker run -it --network=host --device=/dev/kfd --device=/dev/dri --group-add video --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -v $HOME/dockerx:/dockerx -v /data:/data'
drun rocm/pytorch:rocm3.0_ubuntu16.04_py3.6_pytorch
``` 

```Python
import torch
torch.zeros(10).cuda()
tensor([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], device='cuda:0')
```

**Comment by ppanchad-amd** [comment]
Hi @sebpuetz. Do you still need assistance with this ticket? If not, please close the ticket. Thanks!

**Comment by sebpuetz** [comment]
I have not used ROCm in a long time, whether you need to fix something about ROCm 3.0 is up to you :shrug:
